app/assets/javascripts/discourse/app/components/topic-list/list.gjs

```diff
@@ -180,6 +184,16 @@ export default class TopicList extends Component {
     });
   }
 
+  @action
+  didInsertTopicList() {
+    this.prefetch.clearPrefetchedTopics();
+  }
+
+  @action
+  willDestroyTopicList() {
+    this.prefetch.clearPrefetchedTopics();
+  }
+
```

Comment:
Modifiers should only be used for DOM-related things. Since this doesn't relate to the DOM, I think `constructor`/`willDestroy` would be a better fit?